src/engine/SpatialJoin.cpp

```diff
@@ -184,8 +184,9 @@ string SpatialJoin::getDescriptor() const {
       return absl::StrCat("MaxDistJoin ", left, " to ", right, " of ",
                           config.maxDist_, " meter(s)");
     } else if constexpr (std::is_same_v<T, SpatialJoinConfig>) {
-      return absl::StrCat("Spatial Join ", left, " to ", right, " of type ",
-                          config.joinType_);
+      return absl::StrCat(
+          "Spatial Join ", left, " to ", right, " of type ",
+          SpatialJoinTypeString[static_cast<int>(config.joinType_)]);
```

Comment:
Since there is another comment anyway, here is a suggestion for a more idiomatic descriptor:

`Spatial Join of ?a and ?b using intersects`